src/js/background/assignManager.js

```diff
@@ -536,7 +555,270 @@ const assignManager = {
     for (const identity of identities) {
       browser.contextMenus.remove(identity.cookieStoreId);
     }
-  }
+  },
+
+  async initSync() {
```

Comment:
Just a quick thought: since the `assignManager.js` is starting to grow quite large and "synchronization" isn't exactly part of "assigning", it might make sense to move the sync functionality into its own file.

Reply:
Yeah, that is definitely on the list of things to do.